python/cuml/linear_model/linear_regression_mg.pyx

```diff
@@ -0,0 +1,620 @@
+
+        X_ptr = X.ctypes.data
+        y_ptr = y.ctypes.data
+        gpu_ids_ptr = np.array(gpu_ids).ctypes.data
+        coef_ptr = self.coef_.ctypes.data
+
+        cdef float intercept32
+        cdef double intercept64
+
+        if self.gdf_datatype.type == np.float32:
+            olsFitSPMG(<float*>X_ptr,
+                       <int> n_rows,
+                       <int> n_cols,
+                       <float*>y_ptr,
+                       <float*>coef_ptr,
+                       <float*>&intercept32,
+                       <bool>self.fit_intercept,
+                       <bool>self.normalize,
+                       <int*>gpu_ids_ptr,
+                       <int>n_gpus)
+
+            self.intercept_ = intercept32
+
+        else:
+            olsFitSPMG(<double*>X_ptr,
+                       <int> n_rows,
+                       <int> n_cols,
+                       <double*>y_ptr,
+                       <double*>coef_ptr,
+                       <double*>&intercept64,
+                       <bool>self.fit_intercept,
+                       <bool>self.normalize,
+                       <int*>gpu_ids_ptr,
+                       <int>n_gpus)
+
+            self.intercept_ = intercept64
+
+        return self
+
+
+    def _predict_spmg(self, X, gpu_ids):
+
+        n_gpus = len(gpu_ids)
+
+        if (not np.isfortran(X)):
+            X = np.array(X, order='F')
+
+        n_rows = X.shape[0]
+        n_cols = X.shape[1]
+
+        if (n_rows < 1):
+            msg = "X must have a row."
+            raise TypeError(msg)
+
+        if (n_cols < 1):
+            msg = "X must have a column."
+            raise TypeError(msg)
+
+        pred = np.zeros(n_rows, dtype=X.dtype)
+
+        cdef uintptr_t X_ptr, pred_ptr, gpu_ids_ptr, coef_ptr
+
+        self.gdf_datatype = X.dtype
+
+        X_ptr = X.ctypes.data
+        pred_ptr = pred.ctypes.data
+        gpu_ids_ptr = np.array(gpu_ids).ctypes.data
+        coef_ptr = self.coef_.ctypes.data
+
+        if self.gdf_datatype.type == np.float32:
+            olsPredictSPMG(<float*>X_ptr,
+                           <int> n_rows,
+                           <int> n_cols,
+                           <float*>coef_ptr,
+                   <float>self.intercept_,
+                           <float*>pred_ptr,
+                           <int*>gpu_ids_ptr,
+                           <int>n_gpus)
+
+        else:
+            olsPredictSPMG(<double*>X_ptr,
+                           <int> n_rows,
+                           <int> n_cols,
+                           <double*>coef_ptr,
+                   <double>self.intercept_,
+                           <double*>pred_ptr,
+                           <int*>gpu_ids_ptr,
+                           <int>n_gpus)
+
+        return pred
+
+
+    def _fit_dask(self, alloc_info, params):
+
+        self.fit_intercept = params['fit_intercept']
+        self.normalize = params['normalize']
+
+        if alloc_info[0][0]['typestr'] == '<f8':
```

Comment:
This looks like a typecheck, is the object in some type of dict from JSON or something that you need to use strings?

Reply:
`__cuda_array_interface__ ` that we sent from Dask cuML

Reply:
👍